packages/block-library/src/navigation/edit/index.js

```diff
@@ -283,6 +291,15 @@ function Navigation( {
 	const isEntityAvailable =
 		! isNavigationMenuMissing && isNavigationMenuResolved;
 
+	// Migrate numerical ID-based ref to
+	// string slug-based ref.
+	useEffect( () => {
+		if ( isEntityAvailable && isNumeric( ref ) ) {
+			setRef( navigationMenu?.slug );
+			__unstableMarkAutomaticChange();
+		}
+	}, [ isEntityAvailable, ref, navigationMenu ] );
```

Comment:
@scruffian @draganescu This is where we're making an auto update to move from postId to slug.

Also we mark as an automatic change.

This should make the post _content_ dirty (not the entity).